fe/fe-core/src/main/java/com/starrocks/connector/iceberg/cost/IcebergStatisticProvider.java

```diff
@@ -49,13 +53,23 @@
 
 public class IcebergStatisticProvider {
     private static final Logger LOG = LogManager.getLogger(IcebergStatisticProvider.class);
+    private IcebergCatalog icebergcatalog;
 
     public IcebergStatisticProvider() {
     }
 
+    public IcebergStatisticProvider(IcebergCatalog icebergCatalog) {
+        this.icebergcatalog = icebergCatalog;
+    }
+
     public Statistics getTableStatistics(IcebergTable icebergTable, ScalarOperator predicate,
                                          Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
-        LOG.debug("Begin to make iceberg table statistics!");
+        if (icebergcatalog instanceof CachingIcebergCatalog) {
```

Comment:
maybe you should abstract an IcebergStatisticProvider interface. The logic of the current two implementation classes is completely different.